HW0/P4/P4.py

```diff
@@ -30,47 +29,112 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    trajectory_filename =  "P4_trajectory.txt"
+    s_true = np.loadtxt(trajectory_filename,delimiter=',')
+    x_coords = s_true[:,0]
+    y_coords = s_true[:,1]
+    z_coords = s_true[:,2]
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    measurements_filename = "P4_measurements.txt"
+    measurements = np.loadtxt(measurements_filename,delimiter=',')
+    C = np.matrix(np.zeros((3,6)))
+    C[0,0] = rx
+    C[1,1] = ry
+    C[2,2] = rz
+    C_inv = C.getI()
+    x_approx = (C_inv * measurements.T).T
+    ax.plot(x_coords, y_coords, z_coords,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
-
     # Initial conditions for s0
+    A = np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,(1 - c * dt),0,0],[0,0,0,0,(1 - c * dt),0],[0,0,0,0,0,(1 - c * dt)]],dtype=float)
```

Comment:
Good handling of matrices in python!
You can also use sparse numpy arrays with np.diag()